libraries/BlockDevices/SDCardBlockDevice.cpp

```diff
@@ -404,8 +403,8 @@ int SDCardBlockDevice::write(const void *buffer, bd_addr_t add, bd_size_t _size)
          uint32_t num_of_blocks = (_size / write_block_size);
          uint32_t start_block_number = (add / write_block_size);
          rv = FSP_SUCCESS;
-         for(int i = 0; i < num_of_blocks && rv == FSP_SUCCESS; i++) {
-            rv = R_SDHI_Write (&ctrl, (uint8_t *)(buffer + (i * write_block_size)), start_block_number + i, 1);
+         for(uint32_t i = 0; i < num_of_blocks && rv == FSP_SUCCESS; i++) {
+            rv = R_SDHI_Write (&ctrl, (uint8_t *)((uint8_t *)buffer + (i * write_block_size)), start_block_number + i, 1);
```

Comment:
It is just a matter of personal taste but I would prefer this way without the double cast (to me is clearer)
```suggestion
            uint8_t *buf = (uint8_t *)buffer;
            rv = R_SDHI_Write (&ctrl, buf + (i * write_block_size), start_block_number + i, 1);
```